internal/venice-client-common/src/main/java/com/linkedin/venice/compression/GzipCompressor.java

```diff
@@ -19,11 +19,12 @@ public GzipCompressor() {
   @Override
   public byte[] compress(byte[] data) throws IOException {
     ByteArrayOutputStream bos = new ByteArrayOutputStream();
-    try (GZIPOutputStream gos = new GZIPOutputStream(bos)) {
-      gos.write(data);
-      gos.finish();
-      return bos.toByteArray();
-    }
+    ReusableGzipOutputStream out = GzipPool.forStream(bos);
```

Comment:
Is this not tryable?

Reply:
Yes, its going to just reset it for reuse.

Reply:
Sure, will add a try-finally block to reset.